includes/class-wp-smime.php

```diff
@@ -229,4 +239,122 @@ public static function filterContentType ( $content_type ) {
 
         return $content_type . $parameters;
     }
+
+    /**
+     * Securely deletes one or more files by overwriting them.
+     * Tries to handle possible errors gracefully to avoid content disclosure.
+     *
+     * This is by no means foolproof or highly secure, especially when used in a shared hosting scenario.
+     * It's rather an attempt to make the best out of a very sub-optimal situation.
+     *
+     * @param array $files
+     */
+    private static function safeDelete( ...$files ) {
+
+        foreach ( $files as $f ) {
+
+            // clear stat cache to avoid falsely reported file status
+            clearstatcache();
+
+            if ( ! file_exists( $f ) || ! is_file( $f ) ) {
+                // nothing we can do about this
+                error_log( "File path is invalid!" );
+                continue;
+            }
+            if ( ! is_writable( $f ) ) {
+                error_log( "File is not writable!" );
+
+                // cant overwrite, try to delete it
+                if ( ! unlink( $f ) ) {
+                    error_log( "Could not delete file!" );
+                }
+                continue;
+            }
+
+            // try to delegate the work to coreutils 'shred' for better performance
+            // exists on almost all linux systems
+            // https://www.gnu.org/software/coreutils/manual/html_node/shred-invocation.html#shred-invocation
+            $handle = popen( "shred -fuz {$f}", "r" );
```

Comment:
You're right. I should've definitely known better. Probably ignored that because currently there is no user modifiable input, but of course it should be properly escaped.

Agree with using the long options too.